src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -17631,5 +17744,420 @@ public void RefFieldTypeAllowsRefLike()
+                //     T P1
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(3, 5)
+                );
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void AnonymousDelegateType_01_ActionDisallowsRefStruct(bool s2IsRefStruct)
+        {
+            var src = @"
+class C
+{
+    static void Main()
+    {
+        Test1(new S1());
+    }
+    
+    static void Test1<T>(T x) where T : allows ref struct
+    {
+        var d = Helper<T>.Test1;
+        System.Console.Write(d.GetType());
+        System.Console.Write("" "");
+        d(x, new S2());
+    }
+}
+
+class Helper<T> where T : allows ref struct
+{
+    public static void Test1(T x, S2 y)
+    {
+        System.Console.Write(""Test1"");
+        System.Console.Write("" "");
+        System.Console.Write(typeof(T));
+    }
+}
+
+ref struct S1 {}
+
+" + (s2IsRefStruct ? "ref " : "") + @"struct S2 {}
+
+namespace System
+{
+    public delegate void Action<T1, T2>(T1 x, T2 y);
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
+
+            var verifier = CompileAndVerify(
+                comp, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? @"<>A`2[S1,S2] Test1 S1" : null,
+                verify: ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Passes : Verification.Skipped,
+                symbolValidator: (m) =>
+                {
+                    foreach (var tp in m.ContainingAssembly.GetTypeByMetadataName("<>A`2").TypeParameters)
+                    {
+                        Assert.True(tp.AllowByRefLike);
+                    }
+                }
+                ).VerifyDiagnostics();
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void AnonymousDelegateType_02_FuncDisallowsRefStruct(bool s2IsRefStruct)
+        {
+            var src = @"
+class C
+{
+    static void Main()
+    {
+        Test1(new S1());
+    }
+    
+    static void Test1<T>(T x) where T : allows ref struct
+    {
+        var d = Helper<T>.Test1;
+        System.Console.Write(d.GetType());
+        System.Console.Write("" "");
+        d(x);
+    }
+}
+
+class Helper<T> where T : allows ref struct
+{
+    public static S2 Test1(T x)
+    {
+        System.Console.Write(""Test1"");
+        System.Console.Write("" "");
+        System.Console.Write(typeof(T));
+        return default;
+    }
+}
+
+ref struct S1 {}
+
+" + (s2IsRefStruct ? "ref " : "") + @"struct S2 {}
+
+namespace System
+{
+    public delegate T2 Func<T1, T2>(T1 x);
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics, options: TestOptions.ReleaseExe);
+
+            var verifier = CompileAndVerify(
+                comp, expectedOutput: ExecutionConditionUtil.IsMonoOrCoreClr ? @"<>F`2[S1,S2] Test1 S1" : null,
```

Comment:
>ExecutionConditionUtil.IsMonoOrCoreClr ? @"<>F`2[S1,S2] Test1 S1" : null [](http://example.com/codeflow?start=38&length=72)

Consider adding a helper method for this pattern, for example `RefFieldTests.IncludeExpectedOutput()`.